### ISSUE TITLE
An Introduction to Reactive Extensions for .NET (Rx.NET)

### ISSUE THREAD

**Issue by ghost** [selagroup/sdp-dec2016-cfp#30]
**Talk type:** Workshop

**Speaker:** [Sathyaish Chakravarthy](https://github.com/selagroup/sdp-dec2016-cfp/issues/27)

**Technical Level:** Advanced

**Abstract:** In an ever-growing, fast changing technology landscape, consumer expectations have increased at a rapid pace. Customers want data in real time. Buyers want tracking information delivered to their mobile phones in the instant, investors want to watch stock information like hawks, and line of business applications rely on mission critical data to be delivered to them instantaneously to facilitate strategic decision-making.

Reactive Extensions for .NET, Rx in short, is a powerful development framework that enables .NET applications to query events in real time. Rx can also be used to make asynchronous calls in declarative style.

Rx builds on top of Microsoft's existing LINQ infrastructure thus empowering developers to apply complex query operators and transformations to event data in real time. This allows for developers to concentrate on composability and business logic while leaving the plumbing in the safe hands of the framework developers.

In this full-day workshop, we are introduced to the capabilities of this powerful development tool.

**Prerequisites:** You _must_ be well-versed with the following techniques:
1. Object oriented programming in C#
2. To be able to write LINQ queries
3. Know how deferred execution works in LINQ. Here's [a video](https://www.youtube.com/watch?v=oZErvocWFL8) that describes it.
4. Know the benefits that accrue out of monadic composition, or from having a fluent API.
5. Must know about most of the operators available on the `IEnumerable<T>` interface that come out-of-the-box in the `System.Core.dll` assembly, i.e. in the BCL.
6. Be familiar with the observer design pattern.

**Hardware/Software Requirements:** It will be nice if I could have three or four lawn tennis or badminton balls to use during the session. I would like to use them to demonstrate how Rx causes a mind-shift in the way we think about sequences. I will return them to you after the session.

**Audio/Video Requirements:**: I would like permission to bring my cameras to record my own talk and put it up on YouTube.
